src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Capture.cs

```diff
@@ -15,91 +15,63 @@ namespace System.Text.RegularExpressions
     /// </summary>
     public class Capture
     {
-        internal string _text;
-        internal int _index;
-        internal int _length;
-
-        internal Capture(string text, int i, int l)
+        internal Capture(string text, int index, int length)
         {
-            _text = text;
-            _index = i;
-            _length = l;
+            Text = text;
+            Index = index;
+            Length = length;
         }
 
-        /*
-         * The index of the beginning of the matched capture
-         */
         /// <summary>
-        /// Returns the position in the original string where the first character of
-        /// captured substring was found.
+        /// The substring to the left of the capture
         /// </summary>
-        public int Index
+        /// <returns></returns>
+        internal string GetLeftSubstring()
         {
-            get
-            {
-                return _index;
-            }
+            return Text.Substring(0, Index);
         }
 
-        /*
-         * The length of the matched capture
-         */
         /// <summary>
-        /// Returns the length of the captured substring.
+        /// The substring to the right of the capture
         /// </summary>
-        public int Length
+        internal string GetRightSubstring()
         {
-            get
-            {
-                return _length;
-            }
+            return Text.Substring(Index + Length, Text.Length - Index - Length);
```

Comment:
How about slicing rather than substringing?  It looks like at least one of the callers then appends the result to a StringBuilder, so if you do it as a slice, you can avoid the intermediary string.

Reply:
Are you talking about Span slicing here?

Reply:
Got it. I think you meant something like this:

`internal string GetLeftSubstring() => Text.AsReadOnlySpan(0, Index).ToString();`